### PR TITLE
Alpha Redshift support

### PR DESCRIPTION
**Pull request by michel-tricot**
Safer change than https://github.com/LiveRamp/jack/pull/146

The way ids are handled in jack make this change dirty. It always assumes that id is an existing column based on Rails convention. Unfortunately there is no such assumption in redshift and supporting this requires a huge refactoring.

Discovered and fixed a potential bug in code generation: discard columns named tbl and id as they may conflict with existing class members

### CODE REVIEW

**Comment by michel-tricot** [comment]
@bpodgursky can you take quick look at it?

File: pom.xml [hunk 1]
```diff
@@ -83,6 +83,12 @@
     </dependency>
 
     <dependency>
+      <groupId>com.amazon.redshift</groupId>
```
Comment (sidoh) [inline pom.xml]: I don't think this is in maven central. Amazon's awful about not putting their stuff in maven central.

Probably have to find a mirror or check in the jar with the project.

Comment (bpodgursky) [inline pom.xml]: we can put it libs-ext-release at repository.liveramp.com if it's not in maven central (we use it here).  checking in jars with the project is pretty awful.

File: src/rb/templates/persistence_impl.erb [hunk 2]
```diff
@@ -209,7 +209,17 @@ public class <%= model_defn.impl_name %> extends AbstractDatabaseModel<<%= model
   @Override
   protected <%= model_defn.model_name %> instanceFromResultSet(ResultSet rs, Set<Enum> selectedFields) throws SQLException {
     boolean allFields = selectedFields == null || selectedFields.isEmpty();
-    return new <%= model_defn.model_name %>(rs.getLong("id"),
+    <% if database_defn.support_missing_id %>
+    long id;
+    try {
+      id = rs.getLong("id");
+    } catch (SQLException e) {
```
Comment (bpodgursky) [inline src/rb/templates/persistence_impl.erb]: catching this as an exception seems pretty ugly.  could we check for the column as described here?

http://stackoverflow.com/questions/3599861/how-can-i-determine-if-the-column-name-exist-in-the-resultset

File: pom.xml (hunk 1 above)
Comment (michel-tricot) [inline pom.xml]: fixed

File: src/rb/templates/persistence_impl.erb (hunk 2 above)
Comment (michel-tricot) [inline src/rb/templates/persistence_impl.erb]: fixed

**Pull request merged**